src/main/java/datawave/microservice/authorization/federation/FederatedAuthorizationService.java

```diff
@@ -0,0 +1,197 @@
+package datawave.microservice.authorization.federation;
+
+import static datawave.microservice.authorization.preauth.ProxiedEntityX509Filter.ENTITIES_HEADER;
+import static datawave.microservice.authorization.preauth.ProxiedEntityX509Filter.ISSUERS_HEADER;
+
+import java.time.Duration;
+
+import org.apache.http.HttpHeaders;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.reactive.function.client.ExchangeStrategies;
+import org.springframework.web.reactive.function.client.WebClient;
+
+import datawave.microservice.authorization.config.AuthorizationsListSupplier;
+import datawave.microservice.authorization.federation.config.FederatedAuthorizationServiceProperties;
+import datawave.microservice.authorization.federation.config.FederatedAuthorizationServiceProperties.RetryTimeoutProperties;
+import datawave.security.authorization.AuthorizationException;
+import datawave.security.authorization.DatawaveUser;
+import datawave.security.authorization.ProxiedUserDetails;
+import datawave.security.authorization.UserOperations;
+import datawave.user.AuthorizationsListBase;
+import datawave.webservice.result.GenericResponse;
+import reactor.core.publisher.Mono;
+import reactor.util.retry.Retry;
+
+public class FederatedAuthorizationService implements UserOperations {
+    private static final Logger log = LoggerFactory.getLogger(FederatedAuthorizationService.class);
+    
+    public static final String INCLUDE_REMOTE_SERVICES = "includeRemoteServices";
+    
+    private FederatedAuthorizationServiceProperties federatedAuthorizationProperties;
+    private final WebClient webClient;
+    private AuthorizationsListSupplier authorizationsListSupplier;
+    
+    public FederatedAuthorizationService(FederatedAuthorizationServiceProperties federatedAuthorizationProperties, WebClient.Builder webClientBuilder,
+                    AuthorizationsListSupplier authorizationsListSupplier) {
+        this.federatedAuthorizationProperties = federatedAuthorizationProperties;
+        // @formatter:off
+        this.webClient = webClientBuilder
```

Comment:
Do we set a timeout anywhere, and if not should we be? What happens if the federated auth service times out?

Reply:
This part is the overall timeout.
```
.block(Duration.ofMillis(retry.getTimeoutMillis()));
```